### PR TITLE
[MRESOLVER-132] Remove synchronization in TrackingFileManager

### PR DESCRIPTION
**Pull request by michael-o**
See motivation in the ticket as well as the call hierarchy: https://issues.apache.org/jira/browse/MRESOLVER-132

### CODE REVIEW

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java [hunk 1]
```diff
@@ -45,35 +42,28 @@
 
     public Properties read( File file )
     {
-        synchronized ( getLock( file ) )
+        FileInputStream stream = null;
+        try
```
Comment (elharo) [inline maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java:46]: could use try with resources here

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java [hunk 2]
```diff
@@ -83,82 +73,61 @@ public Properties update( File file, Map<String, String> updates )
...
-                return props;
-            }
+            LOGGER.warn( "Failed to create parent directories for tracking file {}", file );
+            return props;
+        }
 
-            RandomAccessFile raf = null;
-            FileLock lock = null;
-            try
+        RandomAccessFile raf = null;
```
Comment (elharo) [inline maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java:83]: RandomAccessFile still feels like a strange way to do this unless there's some synchronization related reason here I don't see

**Comment by michael-o** [comment]
@elharo I'd gently reuse my statement here for the Manager: https://github.com/apache/maven-resolver/pull/66#issuecomment-674696089. We can create a separate issue to improve the code.

**Comment by michael-o** [comment]
@dantran Could you also kindly try this one for you. In my opinion this removes unnecessary locks. Use the same project as for the Redisson solution, but w/o Redisson.

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java [hunk 3]
```diff
@@ -45,35 +42,28 @@
...
     public Properties read( File file )
     {
-        synchronized ( getLock( file ) )
+        FileInputStream stream = null;
+        try
         {
-            FileLock lock = null;
-            FileInputStream stream = null;
-            try
+            if ( !file.exists() )
```
Comment (eolivelli) [inline maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java:48]: If we catch NoSuchFileException below we can save this extra metadata check

**Review by eolivelli — APPROVED** [review]
LGTM

I left one little suggestion for an improvement, it could be significant on slow FS and probably more on Windows 
Btw the change is good even in this form

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java (hunk 3 above)
Comment (michael-o) [reply maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java:48]: Same comment as https://github.com/apache/maven-resolver/pull/67#issuecomment-675411087. The intent is to remove sync only. Improvements should be a separate issue.

**Review by eolivelli — APPROVED** [review]
Makes sense.

+1

**Comment by jira-importer** [comment]
Resolve #896

**Comment by jira-importer** [comment]
Resolve #896

**Pull request closed without merging**